automation_tools/baseimage.py

```diff
@@ -0,0 +1,98 @@
+"""Tasks for base image creation and deployment"""
+from __future__ import print_function
+
+import os
+import time
+
+from fabric.api import env, get, put, run, settings
+from re import search, MULTILINE
+
+from six.moves.urllib.parse import urljoin
+
+
+def create_baseimage(os_url, image=None):
+    """Task to create standard base image using OS URL
+
+    The following environment variables affect this command:
+
+    AUTH_KEYS_URL
+        URL location of authorized_keys file
+
+    :param str os_url: URL of OS media to install
+    :param str image: Image name to be created (without extension .img)
+
+    """
+    if not os_url.endswith('/'):
+            os_url += '/'
+
+    # Detect OS version
+    media = run('wget -q -O- {}'.format(urljoin(os_url, 'media.repo')))
+    if media.succeeded:
+        match_name = search(r'^name\s*=\s*(\D*)\s+([\d\.]*)', media, MULTILINE)
+        if match_name:
+            os_ver = match_name.group(2).split('.')[0]
+        else:
+            os_ver = 7
+
+    if not image:
+        # Detect image name
+        comp_id = run('wget -q -O- {}'.format(urljoin(os_url, '../../../../COMPOSE_ID')))
+        if comp_id.succeeded:
+            match_comp = search(r'(\w+)-([\d\.]+)-(?:\w+-)?([\d\.]+)', comp_id)
+            image = match_comp.group(1).lower() + match_comp.group(2).replace('.', '') + '-' + \
+                match_comp.group(3)
+        else:
+            image = 'unknown-{}'.format(str(time.time()).split('.')[0])
+
+    put('misc/base_image_creation/ks_rhel{}_template'.format(os_ver), 'ks.cfg')
+    run('sed -i "s|OS_URL|{}|g" ks.cfg'.format(os_url))
+    run('sed -i "s|ENCRYPT_PASS|$1$xyz$7xHVh4/yhE6P00NIXbWZA/|g" ks.cfg')
+    run('sed -i "s|AUTH_KEYS_URL|{}|g" ks.cfg'.format(os.getenv('AUTH_KEYS_URL')))
+
+    run('virt-install --connect qemu:///system -n {img} -l {url} -w bridge:br0 '
```

Comment:
do we need to use the `bridge:br0` here? if one would liek to run this on his local or cusotm libvirt, he probalby won't have such bridge configured. + i don't think we need it there. 

not a blocker

Reply:
don't we want to include the qemu-guest-agent serial interface here? or shall we leave this on snap-guest?

Reply:
The latter. It has to be left for snap-guest, here we are creating only baseimage that has to have qemu-guest-agent installed (we do it already). Therefore serial interface has to added to cloned VMs and not to baseimage VM. (but could be for nothing, no loss, no gain)